src/storage/data.rs

```diff
@@ -4,6 +4,99 @@ use {Component, EntitiesRes, Storage};
 use storage::MaskedStorage;
 
 /// A storage with read access.
+///
+/// This is just a type alias for a fetched component storage.
+///
+/// The main functionality it provides is listed in the following,
+/// however make sure to also check out the documentation for the
+/// respective methods on `Storage`.
+///
+/// ## Joining storages
+///
+/// `&ReadStorage` implements `Join`, which allows to do
+/// something like this:
+///
+/// ```
+/// # use specs::*;
+/// #
+/// # struct Pos; impl Component for Pos { type Storage = VecStorage<Self>; }
+/// # struct Vel; impl Component for Vel { type Storage = VecStorage<Self>; }
+/// #
+/// # let mut world = World::new(); world.register::<Pos>(); world.register::<Vel>();
+/// # let pos_storage = world.read::<Pos>();
+/// # let vel_storage = world.read::<Vel>();
+/// (&pos_storage, &vel_storage).join()
+/// # ;
+/// ```
+///
+/// This joins the position and the velocity storage, thus only components
+/// that exist in both storages are iterated over.
```

Comment:
I think  the wording here should be changed. `join` iterates over the components of entities that have both components.